raft/broadcast_tables/experimental/lang.cc

```diff
@@ -25,7 +27,99 @@ bool is_broadcast_table_statement(const sstring& keyspace, const sstring& column
 }
 
 future<> execute(service::raft_group0_client& group0_client, const query& query) {
-    throw exceptions::invalid_request_exception{"executing queries on broadcast_kv_store is currently not implemented"};
+    auto group0_cmd = group0_client.prepare_command(service::broadcast_table_query{query});
+    return group0_client.add_entry_unguarded(std::move(group0_cmd));
+}
+
+
+static
+std::pair<lw_shared_ptr<::query::read_command>, dht::partition_range>
+prepare_read_command(service::storage_proxy& proxy, const schema& schema, const bytes& key) {
+    auto slice = partition_slice_builder(schema).build();
+    auto partition_key = partition_key::from_single_value(schema, key);
+    dht::ring_position ring_position(dht::get_token(schema, partition_key), partition_key);
+    auto range = dht::partition_range::make_singular(ring_position);
+    return {make_lw_shared<::query::read_command>(
+            schema.id(),
+            schema.version(),
+            slice,
+            proxy.get_max_result_size(slice),
+            ::query::tombstone_limit(proxy.get_tombstone_limit())
+        ), range};
+}
+
+static
+const atomic_cell_view
+get_atomic_cell(const schema_ptr& schema, mutation& mutation, const bytes& name) {
+    const auto* column_definition = schema->get_column_definition(name);
+    return mutation.partition().clustered_row(*schema, clustering_key::make_empty()).cells().cell_at(column_definition->id).as_atomic_cell(*column_definition);
+}
+
+future<> execute_broadcast_table_query(
+    service::storage_proxy& proxy,
+    const query& query,
+    utils::UUID cmd_id) {
+    return std::visit(overloaded_functor {
+    [&proxy] (const raft::broadcast_tables::select_query& q) -> future<> {
+        const auto schema = db::system_keyspace::broadcast_kv_store();
+        const auto* column_definition = schema->get_column_definition("value");
+
+        // Read mutations
+        const auto [read_cmd, range] = prepare_read_command(proxy, *schema, q.key);
+        auto [rs, _] = co_await proxy.query_mutations_locally(schema, read_cmd, range, db::no_timeout);
+
+        if (rs->partitions().empty()) {
+            co_return;
+        }
+
+        assert(rs->partitions().size() == 1); // In this version only one value per partition key is allowed.
+
+        const auto& p = rs->partitions()[0];
+        auto mutation = p.mut().unfreeze(schema);
+        const auto cell = get_atomic_cell(schema, mutation, "value");
+
+        // TODO return result
+    },
+    [&proxy, cmd_id] (const raft::broadcast_tables::update_query& q) -> future<> {
+        auto _cmd_id = cmd_id;
+        auto& _proxy = proxy;
+
+        const auto schema = db::system_keyspace::broadcast_kv_store();
+
+        // Read mutations
+        const auto [read_cmd, range] = prepare_read_command(proxy, *schema, q.key);
+        auto [rs, _] = co_await proxy.query_mutations_locally(schema, read_cmd, range, db::no_timeout);
```

Comment:
Nit: you use `proxy` before `co_await` and `_proxy` after `co_await`. Kinda weird. Maybe use `_proxy` everywhere?

Also, I generally suggest a different approach: instead of capturing, take these stuff through regular function arguments, then `std::bind_front` immediately. Then I think it's harder to make a mistake. But it's up to you.

Reply:
Thanks for the advice with `std::bind_front`, now it uses `std::bind_front ` instead of lambda capture.